sled-agent/src/bootstrap/http_entrypoints.rs

```diff
@@ -40,6 +44,52 @@ pub(crate) fn api() -> BootstrapApiDescription {
     api
 }
 
+/// Current status of any rack-level operation being performed by this bootstrap
+/// agent.
+#[derive(
+    Clone, Debug, PartialEq, Eq, Hash, Serialize, Deserialize, JsonSchema,
+)]
+#[serde(tag = "status", rename_all = "snake_case")]
+pub enum RackOperationStatus {
+    None,
+    Initializing {
+        id: Uuid,
+    },
+    /// `id` will be none if the rack was already initialized on startup.
+    Initialized {
+        id: Option<Uuid>,
+    },
+    InitializationFailed {
+        id: Uuid,
+        message: String,
+    },
+    InitializationPanicked {
+        id: Uuid,
+    },
+    ResetOnStartup,
+    Resetting {
+        id: Uuid,
+    },
+    /// `id` will be none if the rack was already reset on startup.
+    Reset {
+        id: Option<Uuid>,
+    },
+    ResetFailed {
+        id: Uuid,
+        message: String,
+    },
+    ResetPanicked {
+        id: Uuid,
+    },
+}
+
+#[derive(
+    Clone, Debug, PartialEq, Eq, Hash, Serialize, Deserialize, JsonSchema,
+)]
+pub struct OperationId {
```

Comment:
Nitpick, there's a really small amount of asymmetry here.

It seems like `OperationId` is passed in the result types, but raw `Uuid`s are exposed from `RackOperationStatus`? Should we be doing one of the other everywhere?

Reply:
Do newtypes place nicely with openapi? For some reason I thought they didn't and so squished the two newtypes down to raw `Uuid`s here for the openapi interface, but I didn't actually try it.

I'll give it a shot. If newtypes _don't_ work nicely with openapi, do you have a preference between:

* Keep it as-is
* Remove the newtypes and use raw `Uuid` everywhere
* Change the newtypes to be openapi-friendly (something like `RackInitId { id: Uuid }`)

Reply:
I'm fine with either of:

- Keep it as-is (after all, this was a nitpick. I wanted to prod in case there was a clean fix - but it isn't a big deal).
- Remove the newtypes and use raw Uuid everywhere

Excessive wrapping of the UUID type under an explicit variable name seems like more pain than it's worth.

Reply:
Newtypes are fine! Removed the raw UUIDs in [cfd2ce7](https://github.com/oxidecomputer/omicron/pull/3351/commits/cfd2ce754f38947c56e49d87a60e4963953c0c42)